components/os/os.go

```diff
@@ -1,40 +1,80 @@
 package os
 
-import "github.com/DataDog/test-infra-definitions/components/command"
+import (
+	"fmt"
 
-type Architecture string
-
-const (
-	AMD64Arch = Architecture("x86_64")
-	ARM64Arch = Architecture("arm64")
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
-// The types of OSes that are common
-type Type int
-
-const (
-	UnixType    Type = iota
-	WindowsType Type = iota
-	OtherType   Type = iota
-)
+// Interfaces used by OS components
+type PackageManager interface {
+	Ensure(packageRef string, opts ...pulumi.ResourceOption) (*remote.Command, error)
+}
 
-type AgentVersion struct {
-	Major       string
-	Minor       string // Empty means latest
-	BetaChannel bool
-	PipelineID  string
-	Arch        string
+type ServiceManager interface {
+	// EnsureStarted starts or restarts (may be stop+start depending on implementation) the service if already running
+	EnsureRestarted(serviceName string, triggers pulumi.ArrayInput, opts ...pulumi.ResourceOption) (*remote.Command, error)
 }
 
+// FileManager needs to be added here as well instead of the command package
+
+// OS is the high-level interface for an OS INSIDE Pulumi code
 type OS interface {
-	GetImage(Architecture) (string, error)
-	GetDefaultInstanceType(Architecture) string
-	GetServiceManager() *ServiceManager
-	GetAgentConfigFolder() string
-	GetSSHUser() string
-	GetAgentInstallCmd(AgentVersion) (string, error)
-	GetRunAgentCmd(parameters string) string
-	GetType() Type
-	CreatePackageManager(runner *command.Runner) (command.PackageManager, error)
-	CheckIsAbsPath(string) bool
+	Descriptor() Descriptor
+
+	Runner() *command.Runner
+	FileManager() *command.FileManager
+	PackageManager() PackageManager
+	ServiceManger() ServiceManager
+}
+
+// os is a generic implementation of OS interface
+type os struct {
+	descriptor     Descriptor
+	runner         *command.Runner
+	fileManager    *command.FileManager
+	packageManager PackageManager
+	serviceManager ServiceManager
+}
+
+func (o os) Descriptor() Descriptor {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
var _ OS = &os{}

func (o os) Descriptor() Descriptor {
```